src/screens/book.js

```diff
@@ -6,25 +6,62 @@ import debounceFn from 'debounce-fn'
 import {FaRegCalendarAlt} from 'react-icons/fa'
 import Tooltip from '@reach/tooltip'
 import {useParams} from 'react-router-dom'
-import {useBook} from 'utils/books'
 import {formatDate} from 'utils/misc'
-import {useListItem, useUpdateListItem} from 'utils/list-items'
 import * as mq from 'styles/media-queries'
 import * as colors from 'styles/colors'
 import {Spinner, Textarea, ErrorMessage} from 'components/lib'
 import {Rating} from 'components/rating'
-import {Profiler} from 'components/profiler'
 import {StatusButtons} from 'components/status-buttons'
+import {useSelector, useDispatch} from 'react-redux'
+import {fetchBookById, selectBookById, selectError} from 'reducers/booksSlice'
+import bookPlaceholderSvg from 'assets/book-placeholder.svg'
+import {selectListItemByBookId, updateListItem} from 'reducers/listItemsSlice'
+import {unwrapResult} from '@reduxjs/toolkit'
+
+const loadingBook = {
+  title: 'Loading...',
+  author: 'loading...',
+  coverImageUrl: bookPlaceholderSvg,
+  publisher: 'Loading Publishing',
+  synopsis: 'Loading...',
+  loadingBook: true,
+}
 
 function BookScreen() {
   const {bookId} = useParams()
-  const book = useBook(bookId)
-  const listItem = useListItem(bookId)
+  const dispatch = useDispatch()
+  const book =
+    useSelector(state => selectBookById(state, bookId)) ?? loadingBook
+  const listItem = useSelector(state => selectListItemByBookId(state, bookId))
+  const error = useSelector(selectError)
 
   const {title, author, coverImageUrl, publisher, synopsis} = book
 
+  const getBookById = React.useCallback(async () => {
+    await dispatch(fetchBookById(bookId))
+  }, [bookId, dispatch])
```

Comment:
Should declare this inside an useEffect block without useCallback since this only gets to be used at one spot